homeassistant/components/tesla/sensor.py

```diff
@@ -37,6 +37,7 @@ def __init__(self, tesla_device, controller, config_entry, sensor_type=None):
         self.units = None
         self.last_changed_time = None
         self.type = sensor_type
+        self._device_class = tesla_device.device_class
```

Comment:
This should preferably be decoupled so we don't return the tesla device class directly as a home assistant entity device class. Make a dict constant in this module that maps between the two APIs.

Reply:
@MartinHjelmare Sorry, I'm not sure I understand what you mean. Are you saying something like this done for any attributes in `tesla_device`:
```python
   MAPPING = { "device_class":tesla_device.device_class}
```
with any getters basically returning
```python
   def device_class():
     return MAPPING["device_class"]
```

Reply:
There should be a dict where the keys are the possible tesla device classes and the values are the corresponding home assistant device class.

Then we use that dict in the device class property by passing the current tesla device class and getting back the corresponding home assistant device class.

```py
TESLA_HA_DEVICE_CLASS = {
    "temperature": DEVICE_CLASS_TEMPERATURE,
    ...
}

...

@property
def device_class(self):
    return TESLA_HA_DEVICE_CLASS.get(self.tesla_device.device_class)
```

Reply:
Does it matter that the device_class in the underlying tesla_device is exactly the same strings as HA?

Reply:
No. These are different APIs in different libraries. Either might change independently of the other and we should never couple two different independent APIs so that they become dependent.

Reply:
For what it's worth, the `teslajsonpy` library has a whole set of files that are intended only for HA use. Essentially because it was created as a requirement of HA to create a pypi dependency, it's very strongly linked and wouldn't work outside of HA's use case.  One of the issues it in fact to generalize it enough to be useful outside of HA. Even when it's generalized, there will likely continue to be the HA specific portion as an example of how to use the library.

Reply:
Regardless we should not have a tight coupling of the APIs inside home assistant.